<html>

<head>

<title>657.000 Lab 3</title>
<link rel="stylesheet" type="text/css" href="657.000.css">
<link rel="stylesheet" type="text/css" href="sidenav.css">
<script type="text/javascript" src="sidenav.js"></script>

<body>

<div id="mySidenav" class="sidenav">
<a href="javascript:void(0)" class="closebtn" onclick="closeNav()">&times;</a>
</div>

<div id="main">

<div class="labtitle">Lab 3&ensp;RNA-Seq Data Analysis</div>
<div class="subtitle">657.000&ensp;High-Throughput Sequence Data Handling (SS 2017)</div>

<span id="contents" onclick="openNav()">
<span style="font-size:125%">&#9776;</span>&ensp;Contents
</span>

<div class="usectitle">Setup</div>
<hr>
<ol>
  <li>In your Linux home directory, make a new directory called <kbd>lab3</kbd>. Do all your work in <kbd>lab3</kbd>.
  <li>Copy <kbd>/export/data/657.000/rnaseq/samples.txt</kbd> to your <kbd>lab3</kbd> directory. This file indicates which samples are from which condition.
  <li>Copy all files in the <kbd>/export/data/657.000/rnaseq</kbd> directory that starts with <kbd>run_</kbd> to your <kbd>lab3</kbd> directory. Use <kbd>/export/data/657.000/rnaseq/run_*</kbd> to refer to those files together. These are shell scripts to run various Trinity programs.
  <li>Make a directory called <kbd>assembly</kbd> under <kbd>lab3</kbd>. Copy <kbd>/export/data/657.000/rnaseq/assembly/Trinity.fasta</kbd> to the <kbd>assembly</kbd> directory. This is the transcriptome assembly.
  <li>Make a directory called <kbd>trinotate</kbd> under <kbd>lab3</kbd>. Copy all files in the <kbd>/export/data/657.000/rnaseq/trinotate</kbd> directory to your <kbd>trinotate</kbd> directory. These are scripts and database files required for annotation with Trinotate.
  <li><a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki" target="_blank">Trinity RNA-Seq</a> contains useful information, which can be directly relevant to some questions.
</ol>

<div id="assembly" class="sectitle">Transcriptome assembly check</div>
<hr>
<ol>
  <li>The <i>de novo</i> transcriptome assembly has been done already, with the results in <kbd>Trinity.fasta</kbd>, which should be in your <kbd>assembly</kbd> directory. This FASTA file contains transcript sequences (contigs) assembled from a random subset (0.5%) of Illumina paired-end RNA-seq read files. Normally you would run the assembly with the <kbd>Trinity</kbd> program (using the <kbd>run_trinity</kbd> script), which is very slow.
  <li>Go to the <kbd>assembly</kbd> directory. Calculate basic statistics about the transcriptome assembly in <kbd>Trinity.fasta</kbd> and save it into a file:
<p>
<table class="command"><tr><td><pre>
/export/programs/trinity/util/TrinityStats.pl Trinity.fasta > assembly_stats.txt
</pre></table>
  <li>Look in <kbd>assembly_stats.txt</kbd> and refer to <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki/Transcriptome-Contig-Nx-and-ExN50-stats" target="_blank">Trinity Transcriptome Contig Nx and ExNy Statistics</a>, to answer these questions:
  <ul class="level2">
    <li>How many genes and transcripts are there in the assembly?
    <li>You will see two different sets of statistics. Which one is a more realistic representation of the assembly and why?
  </ul>
</ol>

<div id="annot" class="sectitle">Functional annotation of transcripts</div>
<hr>
<ol>
  <li>We use <a href="http://trinotate.github.io">Trinotate</a> for functional annotation of the transcripts generated by Trinity. First we need to predict coding regions in the Trinity-generated transcripts. Use the command below to run TransDecoder, which will take about 15 minutes. While you wait for this to be done, you can do the next step, which only needs the transcripts.
<p>
<table class="command"><tr><td><pre>
./run_transdecoder
</pre></table>
  <li>Go to the <kbd>trinotate</kbd> directory. Search the Trinity transcripts in Swiss-Prot using DeCypher accelerated boards for faster BLASTX, with the command below. Without access to the boards, you can use the <kbd>search_transcripts_sprot</kbd> script instead, which will be orders of magnitude slower:
<p>
<table class="command"><tr><td><pre>
./search_transcripts_sprot_dc
</pre></table>
  <li>This step can be done only after the <kbd>run_transdecoder</kbd> script finishes. Search the TransDecoder-predicted proteins in Swiss-Prot using DeCypher boards for faster BLASTP. You can use <kbd>search_proteins_sprot</kbd> when you don't have access to the boards.
<p>
<table class="command"><tr><td><pre>
./search_proteins_sprot_dc
</pre></table>
  <li>The next step is to search the TransDecoder proteins for Pfam protein domain hits. Normally you would use the <kbd>search_proteins_pfam</kbd> script for this task, but it will take about 2 hours. The search has been done for you already, with the results in <kbd>TrinotatePFAM.out</kbd>.
  <li>Load all search results, as well as the transcript and protein sequences, into the Trinotate SQLite database:
<p>
<table class="command"><tr><td><pre>
./load_results_into_db
</pre></table>
  <li>Generate an annotation report from the database, where the report (<kbd>Trinotate_report.xls</kbd>) will be a tab-delimited text file. Use this command, which will take about 10 minutes:
<p>
<table class="command"><tr><td><pre>
./output_annotation_report
</pre></table>
  <li>Open the annotation report with the command below and see what information is provided:
<p>
<table class="command"><tr><td><pre>
libreoffice Trinotate_report.xls
</pre></table>

</ol>

<div id="abund" class="sectitle">Transcript abundance estimation</div>
<hr>
<ol>
  <li>We have 6 samples: <kbd>Dry1, Dry2, Dry3, Wet1, Wet2, Wet3</kbd>. Samples <kbd>Dry1</kbd>, <kbd>Dry2</kbd>, and <kbd>Dry2</kbd> are biological replicates from a dry condition, while samples <kbd>Wet1</kbd>, <kbd>Wet2</kbd>, and <kbd>Wet3</kbd> are biological replicates from a wet condition. We now estimate the abundance of transcripts for each replicate sample separately. <li>In <kbd>lab3</kbd>, build a Bowtie2 index from the transcriptome assembly. Make sure the command runs without an error, until you see the &ldquo;Only prepping reference. Stopping now.&rdquo; message at the end.
<p>
<table class="command"><tr><td><pre>
./run_prep_ref
</pre></table>
  <li>In <kbd>lab3</kbd>, estimate transcript abundance by aligning the reads of each replicate to the transcripts using Bowtie2. This shell script will be silent and can take up to 15 minutes to finish:
<p>
<table class="command"><tr><td><pre>
./run_est_trans_abund
</pre></table>
  <li>Look into <kbd>trans_abund.log</kbd> to find, for each replicate, the overall rate of alignment to the transcriptome. This file has 6 sections, each containing messages output while estimating transcript abundance for a sample. Each section starts with &ldquo;-> Estimating transcript abundance for sample ...&rdquo;.
  <li>Check that there are 6 subdirectories corresponding to each replicate under the <kbd>trans_abund</kbd> directory. Collect the abundance estimates for all replicates into a single matrix, and at the same time expand the transcript ids to include annotation strings using the Trinotate report:
<p>
<table class="command"><tr><td><pre>
./run_abund_to_annot_matrix
</pre></table>
  <li>Go into the <kbd>trans_abund</kbd> directory and see the contents of <kbd>trans.counts.annot.matrix</kbd>. The first column indicates transcript IDs, which can have UniProt protein IDs and additional annotation strings. Not all transcripts have been annotated. Count how many transcripts are annotated with the command below, and explain how it works:
<p>
<table class="command"><tr><td><pre>
grep -c '\^' trans.counts.annot.matrix
</pre></table>
</ol>

<div id="DE" class="sectitle">Differential expression analysis</div>
<hr>
 
<ol>
  <li>In <kbd>lab3</kbd>, compare the transcript abundance estimates and perform differential expression (DE) analysis:
<p>
<table class="command"><tr><td><pre>
./run_DE_analysis
</pre></table>
  <li>Go to the <kbd>diff_expr</kbd> directory, and view the contents of <kbd>trans.counts.annot.matrix.Dry_vs_Wet.edgeR.DE_results</kbd>. This file contains the main results of the DE analysis, where the expression levels under the dry condition are compared to those under the wet condition. Answer these questions using this file:
  <ul class="level2">
    <li>DE results are not available for all transcripts. How many transcripts have a fold change value and additional statistical measures?
    <li>What is the minimum log fold change (FC) and its associated FDR? You can use the <kbd>sort</kbd> command, with &ldquo;logFC&rdquo; (column 4) as the numeric sort key:
<p>
<table class="command"><tr><td><pre>
tail -n +2 trans.counts.annot.matrix.Dry_vs_Wet.edgeR.DE_results | sort -nk4 | head -1 
</pre></table>
    <li>What is the maximum log FC? Use the previous command with the sorting part replaced with <kbd>sort -nrk4</kbd> for reverse sort. Record the log FC and FDR values.
    <li>Open the plots of the DE results with the command below. Locate the two points corresponding to the minimum FC and maximum FC in the &ldquo;Volcano plot&rdquo;. Which change is more statistically significant?
<p>
<table class="command"><tr><td><pre>
evince trans.counts.annot.matrix.Dry_vs_Wet.edgeR.DE_results.MA_n_Volcano.pdf
</pre></table>
</ol>

<div id="DEtrans" class="sectitle">Extraction of differentially expressed transcripts</div>
<hr>

<ol>
  <li>We want to extract the transcripts that are <em>significantly</em> differentially expressed (DE) between the two conditions. In <kbd>lab3</kbd>, extract those transcripts that are at least 4-fold DE with a FDR (false discovery rate) less than 0.001:
<p>
<table class="command"><tr><td><pre>
./run_ext_DE_trans
</pre></table>
  <li>In the <kbd>diff_expr</kbd> directory, find two DE results <em>subset</em> files listing the transcripts that are up-regulated in the dry and wet conditions, respectively. Using these files, answer these questions:
  <ul class="level2">
    <li>How many transcripts are up-regulated in the dry condition? How many of these are annotated?
    <li>How many transcripts are up-regulated in the wet condition? How many of these are annotated?
  </ul>
  <li>View the plot of expression levels of DE transcripts in the samples with the command below. How can you visually estimate the relative amounts of up-regulated transcripts in each condition?
<p>
<table class="command"><tr><td><pre>
evince diffExpr.P0.001_C2.matrix.log2.centered.genes_vs_samples_heatmap.pdf
</pre></table>
</table>
  <li>Each condition (dry and wet) has 3 replicate samples. View the plot of correlations among the replicates, generated based on the extracted DE transcripts: 
<p>
<table class="command"><tr><td><pre>
evince diffExpr.P0.001_C2.matrix.log2.sample_cor_matrix.pdf
</pre></table>
  <ul class="level2">
    <li>Ideally, how should the colors be in the correlation plot?
    <li>Of the 6 replicates, which one do you think has the worst quality?
    <li>Which condition (dry or wet) has a better-quality set of replicates? 
  </ul>
</ol>

<div id="cluster" class="sectitle">Clustering of differentially expressed transcripts</div>
<hr>
<ol>
  <li>We now want to see how the DE transcripts cluster together, based on patterns of expression levels. We can cut the previous hierarchical clustering tree in <kbd>diffExpr.P0.001_C2.matrix.log2.centered.genes_vs_samples_heatmap.pdf</kbd> at a certain percentage of the tree height. In <kbd>lab3</kbd>, cut the tree at 60% of its height to generate fixed set of clusters:
<p>
<table class="command"><tr><td><pre>
./run_cluster_DE_trans
</pre></table>
</table>
  <li>Go to the <kbd>diff_expr/diffExpr.P0.001_C2.matrix.RData.clusters_fixed_P_60</kbd> directory, and wiew the cluster plots:
<p>
<table class="command"><tr><td><pre>
evince my_cluster_plots.pdf
</pre></table>
</table>
  <li>Each plot represents a cluster, where a gray line connects expression levels of a transcript in the cluster over the samples. The blue dots and the connecting line show the centroids (averages) of the transcript abundance over all transcripts in the cluster.
  <ul class="level2">
    <li>How many clusters were formed, and how many transcripts were put into each cluster?
    <li>Go back to the <kbd>diff_expr</kbd> directory and display <kbd>clusters_fixed_P_60.heatmap.heatmap.pdf</kbd>. Can you see how the transcripts have been assigned to the clusters?
  </ul>
</ol>
